packages/block-library/src/block/index.js

```diff
@@ -20,3 +21,29 @@ export const settings = {
 };
 
 export const init = () => initBlock( { name, metadata, settings } );
+
+// In order to make sure that blocks retain their full/wide alignments in the editor when converted to synced patterns
+// we need to add the layout support to the core/block block wrapper. This is only needed in the editor as there is
+// no additional wrapper in the front-end so adding here instead of using the usual supports mechanism in block.json.
+function addEditorOnlyLayoutSupportForPatterns( blockSettings ) {
+	if ( blockSettings.name === 'core/block' ) {
+		if ( ! blockSettings.supports ) {
+			blockSettings.supports = {};
+		}
+		blockSettings.supports.layout = {
+			allowSizingOnChildren: true,
```

Comment:
`allowSizingOnChildren` is only relevant for flex type layouts so it won't apply here.